Cargo.toml

```diff
@@ -25,10 +25,11 @@ defmt = "=0.3.5"
 esp32c3-hal = { version = "0.13.0", default-features = false }
 esp32c2-hal = { version = "0.11.0", default-features = false }
 esp32c6-hal = { version = "0.6.0",  default-features = false }
+esp32h2-hal = { git = "https://github.com/esp-rs/esp-hal.git", package = "esp32h2-hal", default-features = false }
 esp32-hal   = { version = "0.16.0", default-features = false }
 esp32s3-hal = { version = "0.13.0", default-features = false }
 esp32s2-hal = { version = "0.13.0", default-features = false }
-smoltcp = { version = "0.10.0", default-features=false, features = ["medium-ethernet", "socket-raw"] }
+smoltcp = { version = "0.10.0", default-features = false, features = ["proto-igmp", "proto-ipv4", "proto-dns", "socket-tcp", "socket-icmp", "socket-udp", "socket-dns", "medium-ethernet", "proto-dhcpv4", "socket-raw", "socket-dhcpv4"] }
```

Comment:
I'm not sure about this change tbh 🤔

Reply:
The spaces, or the features? 🙃

These features should now be enabled on esp-wifi itself: https://github.com/esp-rs/esp-wifi/blob/main/esp-wifi/Cargo.toml#L101

Reply:
If I revert my `smoltcp` change, I can't build `ble` and `embassy_ble` examples (even with `--no-default-features`) because it expects ` one of the following features: proto-ipv4, proto-ipv6, proto-sixlowpan`, and I'm not sure what ould be the best way of fixing it, haha.

Reply:
You should probably enable one of those features _on the example_ then :)

I guess we could also make smoltcp optional if wifi isn't enabled, but that's out of scope for this PR.

Reply:
You are right, I forgot to revert that change, thanks!